solutions/USACO dec18-bucketlist.mdx

```diff
@@ -41,3 +41,48 @@ file_out.close()
 </PySection>
 
 </LanguageSection>
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <algorithm>  
+#include <cstdio>
+using namespace std;
+int event[1001]; // declaring the array globaly to intialize all the elements to 0
+int main() {
+	freopen("blist.in", "r", stdin);
+	freopen("blist.out", "w", stdout); 
+	int maxtime = 1000;
+	int n; cin >> n;
+		
+		
+	for (int i = 0; i < n; i++) { // iterating through n lines to input the data in
+		int start; cin >> start; 
+		int end; cin >> end;
+		int amt; cin >> amt;
+		event[start] += amt; 
+		// simulating the events by adding/subtracting the amount from si and ti
+		event[end] -= amt; // ^
+	}
+		
+		
+	int res = 0, cur = 0; // keeping track of buckets being used starting from 0
+	for (int i = 0; i <= maxtime; i++) {
+		cur += event[i]; // adding the current bucket to event[i]
```

Comment:
I think the comment can be improved -- comments literally explaining what the code does (for example, "adds event[i] to cur") are not very helpful. Ideally the comment should explain why we are doing this step (ie. "At time i, the number of buckets in use changes by `event[i]` so we want to update the number of buckets we are currently using accordingly" or something like that)